tests/http_spec.rs

```diff
@@ -0,0 +1,216 @@
+    let extension = path
+      .extension()
+      .ok_or(anyhow!("not a valid extension"))?
+      .to_str()
+      .ok_or(anyhow!("not a valid Unicode"))?;
+
+    let spec = match extension.to_lowercase().as_str() {
+      "json" => anyhow::Ok(serde_json::from_str(&contents)?),
+      "yaml" => anyhow::Ok(serde_yaml::from_str(&contents)?),
+      _ => Err(anyhow!("only json and yaml are supported")),
+    };
+    anyhow::Ok(spec?)
+  }
+  async fn setup(&self) -> Arc<ServerContext> {
+    let config = Config::from_file_paths([self.config.clone()].iter())
+      .await
+      .ok()
+      .unwrap();
+    let blueprint = Blueprint::try_from(&config).unwrap();
+    let client = Arc::new(MockHttpClient {
+      upstream_mocks: self.upstream_mocks.to_vec(),
+      expected_upstream_requests: self.expected_upstream_requests.to_vec(),
+    });
+    let server_context = ServerContext::new(blueprint, client);
+    Arc::new(server_context)
+  }
+}
+
+#[derive(Clone)]
+struct MockHttpClient {
+  upstream_mocks: Vec<(UpstreamRequest, UpstreamResponse)>,
+  expected_upstream_requests: Vec<UpstreamRequest>,
+}
+#[async_trait::async_trait]
+impl HttpClient for MockHttpClient {
+  async fn execute(&self, req: reqwest::Request) -> Result<Response, anyhow::Error> {
+    // Clone the mocks to allow iteration without borrowing issues.
+    let mocks = self.upstream_mocks.clone();
+
+    // Try to find a matching mock for the incoming request.
+    let mock = mocks
+      .iter()
+      .find(|(mock_req, _)| {
+        let method_match = req.method().as_str()
+          == serde_json::to_string(&mock_req.0.method.clone())
+            .expect("provided method is not valid")
+            .as_str()
+            .trim_matches('"');
+        let url_match = req.url().as_str() == mock_req.0.url.clone().as_str();
+        method_match && url_match
+      })
+      .expect("Mock not found");
+    // Assert upstream request
+    let upstream_request = mock.0.clone();
+    assert!(
+      self.expected_upstream_requests.contains(&upstream_request),
+      "Unexpected upstream request: {:?}",
+      upstream_request
+    );
+
+    // Clone the response from the mock to avoid borrowing issues.
+    let mock_response = mock.1.clone();
+
+    // Build the response with the status code from the mock.
+    let status_code = reqwest::StatusCode::from_u16(mock_response.0.status)?;
+    let mut response = Response { status: status_code, ..Default::default() };
+
+    // Insert headers from the mock into the response.
+    for (key, value) in mock_response.0.headers {
+      let header_name = HeaderName::from_str(&key)?;
+      let header_value = HeaderValue::from_str(&value)?;
+      response.headers.insert(header_name, header_value);
+    }
+
+    // Set the body of the response.
+    response.body = ConstValue::try_from(serde_json::from_value::<Value>(mock_response.0.body)?)?;
+
+    Ok(response)
+  }
+}
+
+#[tokio::test]
+async fn test_body_yaml() {
```

Comment:
We should detect the format and then load the files. Shouldn't write special handlers for file extensions.

Reply:
done